app/stats/views.py

```diff
@@ -1,88 +1,98 @@
-import altair as alt
-from django.http import JsonResponse
-from django.shortcuts import render
+from datetime import date
+
 import altair
 import duckdb
 import pandas as pd
-from datetime import date
+import urllib
+from django.http import HttpResponseRedirect, JsonResponse
+from django.shortcuts import render
+from django.views.decorators.csrf import csrf_exempt
+
+from .forms import StatFilterForm
+
 
 def retrieve_graph_data(request):
     scope = request.GET.get('scope')
-    # tld = request.GET.get('tld')
+    tld = request.GET.get('tld')
 
     if (scope == 'monthly'):
-        data = retrieve_monthly_chart()
+        data = retrieve_monthly_chart(tld)
     else:
-        data = retrieve_annual_chart()
+        data = retrieve_annual_chart(tld)
     
     return JsonResponse(data.to_dict(), safe = False)
 
+@csrf_exempt
 def index(request):
-    return render(request, 'index.html')
+    if request.htmx:
+        # Refresh partial template
+        form = StatFilterForm(request.POST or None)
+
+        if form.is_valid():
+            time_scope = form.cleaned_data.get("time_scope")
+            tld = form.cleaned_data.get("tld")
 
-def retrieve_annual_chart(from_year = 2009, to_year = date.today().year):
-    from_year = str(from_year)
-    to_year = str(to_year)
+        template_name = "partials/greencheck_graph.html"
+        graph_data_url = "/retrieve_graph_data/?" + urllib.parse.urlencode({'scope': time_scope,'tld': tld})
+    else:
+        # Serve the whole page
+        form = StatFilterForm()
+        template_name = "index.html"
+        graph_data_url = "/retrieve_graph_data/?scope=annually"
+    
+    return render(request, template_name, {'graphUrl': graph_data_url, 'form': form})
+
+def retrieve_annual_chart(tld = None):
+    dataset_location = './stats/static/datasets/annual.development.parquet'
+    start_year = '2009'
+    end_year = '2021'
+   
+    if (tld and tld != 'global'):
+        # Query based on Top-Level Domain (TLD)
+        query = "SELECT * FROM '{}' WHERE year >= {} AND year <= {} AND tld = '{}'".format(dataset_location, start_year, end_year, tld)
```

Comment:
it's better to use the special API for handling SQL queries here. You can see examples below

https://docs.python.org/3.10/library/sqlite3.html#sqlite3.Cursor.execute